### PR TITLE
update implementation according to recent changes

### PR DESCRIPTION
**Pull request by meretp**
- update AIPackage according to restrictions to external properties
- update model_data_explainability and model_data_preprocessing to be lists
- rename buildStart and buildEnd in Build Class
- add supplied_by and standard in Artifact
- add profileIdentifier in CreationInfo and update the Enum class accordingly
- make property "to" optional in relationship
- update Dataset according to restrictions to external properties
- add property datasetType
- change type hint for supplied_by to str in security relationships

### CODE REVIEW

File: src/spdx_tools/spdx3/bump_from_spdx2/package.py [hunk 1]
```diff
@@ -37,7 +37,9 @@ def bump_package(
     # package.supplier -> Relationship, suppliedBy?
     print_missing_conversion("package2.supplier", 0, "https://github.com/spdx/spdx-3-model/issues/113")
```
Comment (armintaenzertng) [inline src/spdx_tools/spdx3/bump_from_spdx2/package.py:38]: supplier can now be bumped, too

File: src/spdx_tools/spdx3/bump_from_spdx2/package.py
Comment (armintaenzertng) [inline src/spdx_tools/spdx3/bump_from_spdx2/package.py:1]: While we are at it:
License comments can now be bumped, too (there is a property `licenseComment` on `License`). According to the [description in the 2.3 spec](https://spdx.github.io/spdx-spec/v2.3/package-information/#716-comments-on-license-field), this should go to the `concludedLicense`.
This probably also applies to files and snippets.

**Review by armintaenzertng — COMMENTED** [review]
Thanks, lgtm, but there are likely some bugs we will have to fix down the road, I imagine.
Two comments regarding bumping.

File: src/spdx_tools/spdx3/bump_from_spdx2/package.py
Comment (armintaenzertng) [reply src/spdx_tools/spdx3/bump_from_spdx2/package.py:1]: Good questions, I also suggest opening an issue!

File: tests/spdx3/model/test_tool.py [hunk 2]
```diff
@@ -26,7 +28,8 @@ def test_invalid_initialization():
         Tool(12, 345)
 
     assert err.value.args[0] == [
-        'SetterError Tool: type of argument "spdx_id" must be str; got int instead: 12',
+        'SetterError Tool: type of argument "spdx_id" must be str; got int instead: ' "12",
```
[the remark this answers is not in the document]
Comment (meretp) [reply tests/spdx3/model/test_tool.py]: deleted

File: src/spdx_tools/spdx3/bump_from_spdx2/package.py (hunk 1 above)
Comment (meretp) [reply src/spdx_tools/spdx3/bump_from_spdx2/package.py]: fixed

**Review by armintaenzertng — APPROVED** [review]
Thanks!
Let's merge this and open a separate issue for the conversion of license comments.

**Pull request merged**